### ISSUE TITLE
A question of update partial data

### ISSUE THREAD

**Issue by liuaiping** [Softmotions/ejdb#141]
For example ,my db like this:
{
     "inventory": {
                  "location1":{"x":5, "y":8, "type":"sw"},
                  "location2":{"x":55, "y":58, "type":"sw2"}, 
                  "location3":{"x":535, "y":538, "type":"sw3"} 
     }
}

ex.Now i only want to update the location1.y or "location1.type".  I want to increment update.
Because the value type maybe different. So i can not used the bson_apend_int or bson_apend_string.
But the bson only accept the json data. So i tried the : 

```
const char *path = "inventory.location1";
const char *data = "{"y":90}";
bson * payload = json2bson(data);

bson bq1;
bson_init_as_query(&bq1);
bson_append_start_object(&bq1, "$upsert");
bson_append_bson(&bq1, path, payload);
bson_append_finish_object(&bq1);
bson_finish(&bq1);
....................

But the result is : 
```

  {
       "inventory.location1": { y: 95 }
  }

   what's wrong?

**Comment by elendilm** [comment]
What your code does is pass a query like
{ "$upsert": { "inventory.location1": { "y": 90} } }
which will replace the inventory.location1 entirely with { "y": 90 }. I couldn't understand how you managed to get the result
{ "inventory.location1": { "y": 95 } } 
-----------------

If you want only to update location1.y, then you pass a query like
{ "$set": { "inventory.location1.y": 90 } }

You replace the entire location1  with a query like
{ "$set": { "inventory.location1": { "y": 90} } }

If you only want to increment location1.y with 5, then you pass a query like
{ "$inc": { "inventory.location1.y": 5 } }

If you want to increment location1.y with 5 and update location1.type, then you pass a query like
{ "$inc": { "inventory.location1.y": 5 },  "$set": { "inventory.location1.type": "xy" } }

Why you used $upsert is beyond me. Sorry if this doesn't answer as I couldn't figure out what you were trying to say.

**Comment by liuaiping** [comment]
Thanks @elendilm. Maybe my codes were not clear. I wanted to write a API to write data to DB.  The language is C. So i couldn't use bson_append_int or bson_append_string etc. Because the type would many types.(Right?) So i want to use bson_append_bson() and the data should be a json (like: {"y":90}). I don't known how can i do that.

**Comment by adamansky** [comment]
Closed in favour to ejdb 2.0 https://github.com/Softmotions/ejdb/blob/master/README.md
